lib/floe/workflow/choice_rule/data.rb

```diff
@@ -156,6 +157,28 @@ def parse_path(field_name, payload)
           missing_field_error!(field_name) unless value
           wrap_parser_error(field_name, value) { Path.new(value) }
         end
+
+        # parse predicate field at initialization time
+        def parse_field(field_name, payload)
+          value = payload[field_name]
+          return value if correct_type?(value)
+
+          invalid_field_error!(field_name, value, "required to be a #{type || "Boolean"}")
```

Comment:
So we do type || Boolean in two places, would it make sense to set this at initialization time or just check if type is nil

Reply:
fixed. good call